base/src/main/java/org/aya/concrete/Expr.java

```diff
@@ -251,6 +255,34 @@ record BinOpSeq(
     @NotNull ImmutableSeq<NamedArg> seq
   ) implements Expr {}
 
+  record FaceExpr(@NotNull SourcePos sourcePos) implements Expr {}
+
+  /** partial element */
```

Comment:
Needs update, see https://github.com/ice1000/guest0x0/compare/6b0196b6b21ec6a9086edb97802f21bbafbd10f7...415744a9ed74e825bd88185debfd6999d131c35c

Reply:
woww!! with this change I finally understand the code below.
![image](https://user-images.githubusercontent.com/9902643/186119335-e494ce6d-2f27-4345-a992-a5ab48ffab78.png)

Reply:
No I don't understand

Reply:
Bc type checking guarantees both sides to be partial elements of the same type (and have the same set of faces). It remains to make sure all faces are the same. So, we just traverse the clauses of one side and see if the other side reduces the same way.

Reply:
Thank you for the explanation. But my problem was the `n` here is also a partial element (it should be the right-hand side of another partial element). This eventually compares `u` with `{| u |}`. Am I wrong? 😭

Reply:
No, you're right. I'll fix it in guest. Thanks for the reminder!